test/Yesod/Paginator/WidgetsSpec.hs

```diff
@@ -52,6 +53,27 @@ spec = withApp $ do
             , "</ul>"
             ]
 
+      it "includes all parameters with the same name" $ do
```

Comment:
Is it overkill to test this functionality in an E2E test? I guess another way would be to test the specific formatting function, but that would require either exposing it or sticking it in an internal module or something.

Reply:
> but that would require either exposing it or sticking it in an internal module or something

I'm fine with either of these. I tend to just export it, under an `-- * Exported for testing` Haddock heading so it's clear it's not part of the versioned package API.

Reply:
Ok, cool, I'll make it a pure function and export it to avoid the unnecessarily complex E2E test.